### DISCUSSION TITLE
Use pinia outside components reusing the code

### DISCUSSION THREAD

**Discussion by efimatica** [vuejs/pinia#1158] [Help and Questions]
I followed the `services` pattern so I have all the interactions with the API in isolated javascript files, like:

`multimediaService.js`

```
import axios from 'axios'

export const useMultimediaCreateService = (data) => {
  return axios
    .post(`${API}/property/${propertyId}/photo`, data)
    .then((res) => res.data)
}
```

As you can see I need the propertyId variable which is stored on a pinia store.
In my use cases, almost for every API request I need that variable.
So my idea was do the following:

```
import axios from 'axios'
import { usePropertyStore } from '@/core/stores/propertyStore'

// This is failing because it's called before App creates Pinia.
const propertyStore = usePropertyStore()
const propertyId = propertyStore.propertyId

export const useMultimediaCreateService = (data) => {
  return axios
    .post(`${API}/property/${propertyId}/photo`, data)
    .then((res) => res.data)
}
```

But this 2 lines are failing because I'm using the store before App created Pinia.
If I move these 2 lines inside the `useMultimediaCreateService` works because this function is called after App creates Pinia, a SFC will import that service and call it.

The thing is, I have a lot of thin methods like `useMultimediaCreateService` so I cannot duplicate the code and paste these 2 lines of every method...
I can implement a method to extract that `propertyId` variable but I will have to call this method on all methods inside services files and it's a bit frustrating 😢

Is there another way to achieve that I want?

**Comment by dreambo8563** [comment]
@efimatica, I am not sure if this is you wanted. 

sample code for reusable service within pinia store:

```ts

export const CounterService = () => {
  const counterStore = useCounterStore();

  // reusable method
  const ajax = () => {
    return Promise.resolve(counterStore.counter);
  };

  return {
    ajax,
  };
};
```
2. another way. you can read the doc https://pinia.vuejs.org/cookbook/composing-stores.html#shared-actions. wrap your shared methods as actions. then you can use store there
